code/game/objects/items/tools/weldingtool.dm

```diff
@@ -39,6 +39,7 @@
 	var/progress_flash_divisor = 10
 	var/burned_fuel_for = 0	//when fuel was last removed
 	var/light_intensity = 2
+	var/functional = TRUE //Used by the cyborg welders to determine if they're been smashed by a nightmare
```

Comment:
Rather than use a boolean value, store the time that the welding tool will be able to be used again at. This will allow for lighteater_act to be called on it while it is already functional without causing issues

Reply:
Done